synapse/lib/cell.py

```diff
@@ -1185,6 +1273,102 @@ def walkpath(path):
         walkpath(self.backdirn)
         return backups
 
+    async def iterBackupArchive(self, name, user):
+
+        path = self._reqBackDirn(name)
+        cellguid = os.path.join(path, 'cell.guid')
+        if not os.path.isfile(cellguid):
+            mesg = 'Specified backup path has no cell.guid file.'
+            raise s_exc.BadArg(mesg=mesg)
+
+        link = s_scope.get('link')
+        linkinfo = link.getSpawnInfo()
+
+        await self.boss.promote('backup:stream', user=user)
+
+        ctx = multiprocessing.get_context('spawn')
+        done = ctx.Queue()
+
+        proc = None
+        mesg = 'Streaming complete'
+
+        def getproc():
+            proc = ctx.Process(target=_iterBackupProc, args=(path, linkinfo, done))
+            proc.start()
+            return proc
+
+        try:
+            proc = await s_coro.executor(getproc)
+
+            def waitproc():
+                proc.join()
+                return done.get()
+
+            if await s_coro.executor(waitproc):
+                await link.fini()
+
+        except (asyncio.CancelledError, Exception) as e:
+
+            if not isinstance(e, asyncio.CancelledError):
+                logger.exception('Error during backup streaming.')
+
+            if proc:
+                await s_coro.executor(done.put, False)
+                proc.terminate()
+
+            mesg = repr(e)
+            raise
+
+        finally:
+            raise s_exc.DmonSpawn(mesg=mesg)
+
+    async def iterNewBackupArchive(self, name, user, remove=False):
+
+        await self.runBackup(name)
```

Comment:
`name = await self.runBackup(name)` to allow auto-generation to return a value?

Reply:
Lets make name an optional argument ( just like runBackup() ) to auto-populate a time based name ( here and the telepath endpoint ) and have a test that calls with name=None

Reply:
Updated name to an optional arg